torchgeo/datasets/ssl4eo.py

```diff
@@ -0,0 +1,250 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""Self-Supervised Learning for Earth Observation."""
+
+import os
+import random
+from typing import Callable, Dict, Optional, cast
+
+import matplotlib.pyplot as plt
+import numpy as np
+import rasterio
+import torch
+from torch import Tensor
+
+from .geo import NonGeoDataset
+from .utils import check_integrity, extract_archive
+
+
+class SSL4EOS12(NonGeoDataset):
+    """SSL4EO-S12 dataset.
+
+    The `Self-Supervised Learning for Earth Observation - Sentinel-1/2
+    <https://github.com/zhu-xlab/SSL4EO-S12>`_ dataset is a large-scale mutilmodal
+    multitemporal dataset for unsupervised/self-supervised pre-training in Earth
+    observation.
+
+    The dataset consists of unlabeled patch triplets (Sentinel-1 dual-pol SAR,
+    Sentinel-2 top-of-atmosphere multispectral, Sentinel-2 surface reflectance
+    multispectral) from 251079 locations across the globe, each patch covering
+    2640mx2640m and including four seasonal time stamps.
+
+    If you use this dataset in your research, please cite the following paper:
+
+    * https://arxiv.org/abs/2211.07044
+
+    .. note::
+
+       This dataset can be downloaded using:
+
+       .. code-block:: console
+
+          $ export RSYNC_PASSWORD=m1660427.001
+          $ rsync -av rsync://m1660427.001@dataserv.ub.tum.de/m1660427.001/ .
+
+       The dataset is about 1.5 TB when compressed and 3.7 TB when uncompressed, and
+       takes roughly 36 hrs to download, 1 hr to checksum, and 12 hrs to extract.
+
+    .. versionadded:: 0.5
+    """
+
+    size = 264
+
+    metadata = {
+        "s1": {
+            "filename": "s1.tar.gz",
+            "md5": "51ee23b33eb0a2f920bda25225072f3a",
+            "bands": ["VV", "VH"],
+        },
+        "s2c": {
+            "filename": "s2_l1c.tar.gz",
+            "md5": "b4f8b03c365e4a85780ded600b7497ab",
+            "bands": [
+                "B1",
+                "B2",
+                "B3",
+                "B4",
+                "B5",
+                "B6",
+                "B7",
+                "B8",
+                "B8A",
+                "B9",
+                "B11",
+                "B12",
+            ],
+        },
+        "s2a": {
+            "filename": "s2_l2a.tar.gz",
+            "md5": "85496cd9d6742aee03b6a1c99cee0ac1",
+            "bands": [
+                "B1",
+                "B2",
+                "B3",
+                "B4",
+                "B5",
+                "B6",
+                "B7",
+                "B8",
+                "B8A",
+                "B9",
+                "B11",
+                "B12",
+            ],
+        },
+    }
+
+    def __init__(
+        self,
+        root: str = "data",
+        split: str = "s2c",
+        seasons: int = 1,
```

Comment:
As far as I understand "seasons" decides how many seasonal patches to sample and it does so randomly. Would you also want to give control over the seasons in the sense if a user only wants to sample from a particular or a subset of the four seasons?

Reply:
That could be a feature worth adding later, but I'm trying to keep things simple until someone actually needs that functionality.